### ISSUE TITLE
Provisioning fails with chef api error

### ISSUE THREAD

**Issue by thommay** [chef/chef-provisioning#394]
Running `chef provision spincycle --sync -n spincycle_jenkins` in https://github.com/chef/spincycle-cookbooks results in this error:
```  ================================================================================
    Error executing action `setup` on resource 'machine[spincycle_jenkins]'
    ================================================================================

    Net::HTTPServerException
    ------------------------
    chef_client[spincycle_jenkins] (basic_chef_client::block line 132) had an error: Net::HTTPServerException: 400 "Bad Request"

    Resource Declaration:
    ---------------------
    # In /Users/thom/work/cookbooks/spincycle-cookbooks/provision/recipes/default.rb

     15: machine context.node_name do
     16:   machine_options bootstrap_options: {iam_instance_profile: { name: 'spincycle_provisioner'},
     17:                                       key_name: node['ssh_key'],
     18:                                       image_id: 'ami-b33dccf7',
     19:                                       instance_type: 'm3.medium',
     20:                                       security_group_ids: ['sg-6c37a109']
     21:   },
     22:     transport_options: {ssh_options: { use_agent: true}},
     23:     convergence_options: context.convergence_options
     24:   action :setup
     25: end
     26:

    Compiled Resource:
    ------------------
    # Declared in /Users/thom/work/cookbooks/spincycle-cookbooks/provision/recipes/default.rb:15:in `from_file'

    machine("spincycle_jenkins") do
      action [:setup]
      retries 0
      retry_delay 2
      default_guard_interpreter :default
      chef_server {:chef_server_url=>"https://api.opscode.com/organizations/spincycle", :options=>{:client_name=>"thom", :signing_key_filename=>"/Users/thom/.chef/thom.pem"}}
      driver "aws:default:us-west-1"
      machine_options {:bootstrap_options=>{:iam_instance_profile=>{:name=>"spincycle_provisioner"}, :key_name=>"spincycle_jenkins", :image_id=>"ami-b33dccf7", :instance_type=>"m3.medium", :security_group_ids=>["sg-6c37a109"]}, :transport_options=>{:ssh_options=>{:use_agent=>true}}, :convergence_options=>{:chef_server=>"https://api.opscode.com/organizations/spincycle", :chef_config=>"# SSL Settings:\nssl_verify_mode :verify_peer\n\n# Policyfile Settings:\nuse_policyfile true\npolicy_document_native_api true\n\npolicy_group \"spincycle\"\npolicy_name \"spincycle-jenkins\"\n\n"}}
      declared_type :machine
      cookbook_name :provision
      recipe_name "default"
    end

Error: Chef failed to converge: machine[spincycle_jenkins] (provision::default line 15) had an error: Net::HTTPServerException: chef_client[spincycle_jenkins] (basic_chef_client::block line 132) had an error: Net::HTTPServerException: 400 "Bad Request"
Reason: (Net::HTTPServerException) HTTP 400 Bad Request: Since Server API v1, all keys must be updated via the keys endpoint.
```

```
17:38 ~/work/cookbooks/spincycle-cookbooks % chef -v                   [master]
Chef Development Kit Version: 0.7.0.rc.0
chef-client version: 12.4.1
berks version: 3.2.4
kitchen version: 1.4.1
17:38 ~/work/cookbooks/spincycle-cookbooks % chef gem list G provis    [master]
chef-provisioning (1.2.1, 1.2.0)
chef-provisioning-aws (1.3.0, 1.2.1)
chef-provisioning-azure (0.3.2)
chef-provisioning-fog (0.13.2)
chef-provisioning-vagrant (0.8.3)
```

**Comment by tyler-ball** [comment]
Hey @thommay - can you gist the output of the full stacktrace?  The error text should show the location of it.

**Comment by thommay** [comment]
```
--- REQUEST DATA ----
PUT /organizations/spincycle/clients/spincycle_jenkins
content-type: application/json
accept: application/json
accept-encoding: gzip;q=1.0,deflate;q=0.6,identity;q=0.3
x-ops-sign: algorithm=sha1;version=1.0;
x-ops-userid: thom
x-ops-timestamp: 2015-07-22T16:49:44Z
x-ops-content-hash: ZClYWv1/FfT2fY/GzV2k+aiyU4c=
host: api.opscode.com:443
x-ops-server-api-version: 1
x-remote-request-id: 9c15d564-355b-43f2-9afb-933794f183ff
content-length: 612
x-chef-version: 12.4.1
user-agent: Chef Knife/12.4.1 (ruby-2.1.6-p336; ohai-8.5.0; x86_64-darwin12.0; +http://opscode.com)
connection: close

{"name":"spincycle_jenkins","clientname":"spincycle_jenkins","admin":false,"validator":false,"chef_type":"client","orgname":"spincycle","public_key":"ELIDED"}

--- RESPONSE DATA ---
400 Bad Request
server: ngx_openresty
date: Wed, 22 Jul 2015 16:49:09 GMT
content-length: 83
connection: close
x-ops-server-api-version: {"min_version":"0","max_version":"1","request_version":"1","response_version":"1"}
x-ops-api-info: flavor=hec;version=12.0.0;oc_erchef=12.1.0


{"error":["Since Server API v1, all keys must be updated via the keys endpoint. "]}

/opt/chefdk/embedded/lib/ruby/2.1.0/net/http/response.rb:119:in `error!'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.1/lib/chef/http.rb:145:in `request'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.1/lib/chef/http.rb:118:in `put'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/cheffish-1.2.1/lib/cheffish/actor_provider_base.rb:19:in `block in create_actor'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.1/lib/chef/mixin/why_run.rb:52:in `call'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.1/lib/chef/mixin/why_run.rb:52:in `add_action'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.1/lib/chef/provider.rb:175:in `converge_by'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/cheffish-1.2.1/lib/cheffish/actor_provider_base.rb:18:in `create_actor'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/cheffish-1.2.1/lib/chef/provider/chef_client.rb:20:in `block in <class:ChefClient>'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.1/lib/chef/provider/lwrp_base.rb:160:in `instance_eval'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.1/lib/chef/provider/lwrp_base.rb:160:in `block in action'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.1/lib/chef/provider.rb:140:in `run_action'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.1/lib/chef/resource.rb:584:in `run_action'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.1/lib/chef/runner.rb:49:in `run_action'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.1/lib/chef/runner.rb:81:in `block (2 levels) in converge'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.1/lib/chef/runner.rb:81:in `each'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.1/lib/chef/runner.rb:81:in `block in converge'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.1/lib/chef/resource_collection/resource_list.rb:83:in `block in execute_each_resource'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.1/lib/chef/resource_collection/stepable_iterator.rb:116:in `call'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.1/lib/chef/resource_collection/stepable_iterator.rb:116:in `call_iterator_block'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.1/lib/chef/resource_collection/stepable_iterator.rb:85:in `step'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.1/lib/chef/resource_collection/stepable_iterator.rb:104:in `iterate'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.1/lib/chef/resource_collection/stepable_iterator.rb:55:in `each_with_index'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.1/lib/chef/resource_collection/resource_list.rb:81:in `execute_each_resource'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.1/lib/chef/runner.rb:80:in `converge'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/cheffish-1.2.1/lib/cheffish/basic_chef_client.rb:69:in `block in converge'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/cheffish-1.2.1/lib/cheffish/basic_chef_client.rb:133:in `call'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/cheffish-1.2.1/lib/cheffish/basic_chef_client.rb:133:in `with_chef_config'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/cheffish-1.2.1/lib/cheffish/basic_chef_client.rb:68:in `converge'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/cheffish-1.2.1/lib/cheffish/basic_chef_client.rb:109:in `converge_block'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/chef-provisioning-1.2.1/lib/chef/provisioning.rb:12:in `inline_resource'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/chef-provisioning-1.2.1/lib/chef/provisioning/convergence_strategy/precreate_chef_objects.rb:130:in `create_chef_objects'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/chef-provisioning-1.2.1/lib/chef/provisioning/convergence_strategy/precreate_chef_objects.rb:18:in `setup_convergence'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/chef-provisioning-1.2.1/lib/chef/provisioning/convergence_strategy/install_sh.rb:40:in `setup_convergence'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/chef-provisioning-1.2.1/lib/chef/provisioning/machine/basic_machine.rb:17:in `setup_convergence'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/chef-provisioning-1.2.1/lib/chef/provider/machine.rb:46:in `block in <class:Machine>'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.1/lib/chef/provider/lwrp_base.rb:61:in `instance_eval'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.1/lib/chef/provider/lwrp_base.rb:61:in `recipe_eval_with_update_check'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.1/lib/chef/provider/lwrp_base.rb:46:in `block in action'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.1/lib/chef/provider.rb:140:in `run_action'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.1/lib/chef/resource.rb:584:in `run_action'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.1/lib/chef/runner.rb:49:in `run_action'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.1/lib/chef/runner.rb:81:in `block (2 levels) in converge'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.1/lib/chef/runner.rb:81:in `each'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.1/lib/chef/runner.rb:81:in `block in converge'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.1/lib/chef/resource_collection/resource_list.rb:83:in `block in execute_each_resource'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.1/lib/chef/resource_collection/stepable_iterator.rb:116:in `call'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.1/lib/chef/resource_collection/stepable_iterator.rb:116:in `call_iterator_block'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.1/lib/chef/resource_collection/stepable_iterator.rb:85:in `step'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.1/lib/chef/resource_collection/stepable_iterator.rb:104:in `iterate'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.1/lib/chef/resource_collection/stepable_iterator.rb:55:in `each_with_index'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.1/lib/chef/resource_collection/resource_list.rb:81:in `execute_each_resource'
/opt/chefdk/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.1/lib/chef/runner.rb:80:in `converge'
/opt/chefdk/embedded/apps/chef-dk/lib/chef-dk/chef_runner.rb:38:in `converge'
/opt/chefdk/embedded/apps/chef-dk/lib/chef-dk/command/provision.rb:242:in `run'
/opt/chefdk/embedded/apps/chef-dk/lib/chef-dk/command/base.rb:57:in `run_with_default_options'
/opt/chefdk/embedded/apps/chef-dk/lib/chef-dk/cli.rb:73:in `run'
/opt/chefdk/embedded/apps/chef-dk/bin/chef:25:in `<top (required)>'
/opt/chefdk/bin/chef:73:in `load'
/opt/chefdk/bin/chef:73:in `<main>'
```

**Comment by korishev** [comment]
This is an issue with chef 12.4.1 and 12.4.0, reverting to 12.3.x fixes the issue

**Comment by tyler-ball** [comment]
@korishev and @thommay What version of Chef Server are you using?  I haven't seen this but I always test with chef-zero.  I'll put together a test where I hit Hosted Chef

**Comment by korishev** [comment]
@tyler-ball I'm using Hosted Chef.

**Comment by thommay** [comment]
also hosted, but 12.1.x would exhibit the same behaviour - https://github.com/chef/chef-server/blob/master/omnibus/RELEASE_NOTES.md#api-changes-and-additions

**Comment by thommay** [comment]
This is chef/cheffish#56

**Comment by tyler-ball** [comment]
This fix will be released in cheffish 1.3.1 which will be included in the ChefDK 0.7.0 release
